apps/prairielearn/src/lib/issues.ts

```diff
@@ -1,28 +1,30 @@
-// @ts-check
 import { eachSeries } from 'async';
 import { callbackify } from 'util';
 
 import * as sqldb from '@prairielearn/postgres';
 import { recursivelyTruncateStrings } from '@prairielearn/sanitize';
+import { Variant } from './db-types';
 
-/**
- * @typedef {Object} IssueData
- * @property {number | string} variantId
- * @property {string | null} studentMessage
- * @property {string | null} instructorMessage
- * @property {boolean} manuallyReported
- * @property {boolean} courseCaused
- * @property {Object} courseData
- * @property {Object} systemData
- * @property {number | string | null} authnUserId
- */
+interface IssueForErrorData {
+  variantId: string;
+  studentMessage: string | null;
+  courseData: Record<string, any>;
+  authnUserId: string | null;
+}
+
+interface IssueData extends IssueForErrorData {
+  instructorMessage: string | null;
+  manuallyReported: boolean;
+  courseCaused: boolean;
+  systemData: Record<string, any>;
+}
 
-/** @typedef {Omit<IssueData, 'manuallyReported' | 'courseCaused' | 'instructorMessage' |'systemData'>} IssueForErrorData */
+interface ErrorMaybeWithData extends Error {
+  data?: any;
+}
```

Comment:
A similar type exists in `@prairielearn/error`. Should we export it from that package?

Reply:
It's not exactly the same, though I could have an `Error | ErrorWithData`. Is that ok?

Reply:
That'd work well, yeah.